zulip/zulip/__init__.py

```diff
@@ -1634,7 +1649,7 @@ def move_topic(
         topic: str,
         new_topic: Optional[str] = None,
         message_id: Optional[int] = None,
-        propagate_mode: str = "change_all",
+        propagate_mode: EditPropagateMode = "change_all",
```

Comment:
Can this perhaps just have the `Literal` type directly, inline? That'd be ideal from the perspective of someone looking at this function signature and wanting to understand how to use it.

If that's inconvenient for some reason, then a one-line comment on the `EditPropagateMode` definition, mentioning it's for this method, would be helpful for the reader looking at that definition and wondering what it means.

Reply:
I agree, since we aren't using this type anywhere else.

Reply:
I think having comments relating it to which API calls would be useful, and for other types too, when we introduce those.

For discussion on whether it's worth introducing them as named types, see the discussion thread for details - but in this case the argument is that it's useful to track the proposed mode in an application before sending it, and also mapping it to user-displayed strings.